setup.cfg

```diff
@@ -189,7 +190,9 @@ extend-ignore =
     RST213,
     # RST305 so that ReST substitutions in a different file can be used
     RST305,
-    RST306
+    RST306,
+    # TC003 to allow long messages to be specified outside the exception class
+    TC003
```

Comment:
[TC003](https://github.com/guilatrova/tryceratops/blob/main/docs/violations/TC003.md) seems like a good thing to implement, but would require comparatively large changes. 

Also, it gets applied to builtin exceptions like `TypeError` for which we can't straightforwardly take this approach unless we create a `TypeError` subclass.

Reply:
Wow yeah, I didn't even think about it but it makes a lot of sense. I don't even mind making those TypeError subclasses, honestly, if it reduces duplication.

Reply:
Good idea!  I created these in #1789 and #1790.